src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +86,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        List<String> collect = peopleList.stream()
+                .filter(x -> x.getSex() == Person.Sex.WOMAN && x.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
You can do it in one line
```suggestion
                .flatMap(person -> person.getCats().stream())
```